databases/gardener/gardener.rb

```diff
@@ -45,51 +45,53 @@
 def add_crop(db, crop, bed_number)
   occurence = "plant"
   db.execute("INSERT INTO crops (crop, bed_number) values (?, ?)", [crop, bed_number])
-  log(db, occurence, crop)
+  id_n = db.execute("SELECT crops.id FROM crops WHERE crop=(?)", [crop])
+  id = id_n[0]["id"]
+  log(db, occurence, id)
 end
 
-def add_note(db, crop, note)
-  id = db.execute("SELECT crops.notes_id FROM crops WHERE crop=(?)", [crop]) #are there existing notes for this crop?
+def add_note(db, id, note)
```

Comment:
This feels like notes should be its own table at this point.  If you create notes for your crops then you can create a not that links to the crop table through crop id. Then you could cut out all of the note type searches.